core/src/main/java/org/mule/registry/DefaultRegistryBroker.java

```diff
@@ -7,41 +7,72 @@
 package org.mule.registry;
 
 import org.mule.api.MuleContext;
+import org.mule.api.registry.LifecycleRegistry;
 import org.mule.api.registry.Registry;
 
 import java.util.Collection;
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicReference;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @deprecated as of 3.7.0. This will be removed in Mule 4.0
+ */
+@Deprecated
 public class DefaultRegistryBroker extends AbstractRegistryBroker
 {
-    private TransientRegistry transientRegistry;
-    private List<Registry> registries = new CopyOnWriteArrayList<Registry>();
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultRegistryBroker.class);
+
+    private final List<Registry> registries = new CopyOnWriteArrayList<>();
+    private final AtomicReference<LifecycleRegistry> lifecycleRegistry = new AtomicReference<>(null);
 
     public DefaultRegistryBroker(MuleContext context)
     {
         super(context);
-        transientRegistry = new TransientRegistry(context);
-        registries.add(0, transientRegistry);
     }
 
-    TransientRegistry getTransientRegistry()
-    {
-        return transientRegistry;
-    }
 
     public void addRegistry(Registry registry)
     {
-        registries.add(1, registry);
+        registries.add(0, registry);
+        if (registries.size() > 1 && LOGGER.isWarnEnabled())
+        {
+            LOGGER.warn("A second registry has been registered, however adding registries has been deprecated as of Mule 3.7.0");
+        }
     }
 
     public void removeRegistry(Registry registry)
     {
         registries.remove(registry);
+        if (registry instanceof LifecycleRegistry)
```

Comment:
this is doing the opposite. If the registry removed is cached as the one handling lifecycle, then the reference is nulled.

It's a little trick to minimize contention.